src/Catel.Core/Data/Adapters/ReflectionObjectAdapter.cs

```diff
@@ -0,0 +1,211 @@
+﻿namespace Catel.Data
+{
+    using System;
+    using Catel.Logging;
+    using Catel.Reflection;
+    using Catel.Runtime.Serialization;
+
+    /// <summary>
+    /// Object adapter allowing to customize reflection and property mappings.
+    /// </summary>
+    public class ReflectionObjectAdapter : IObjectAdapter
+    {
+        private static readonly ILog Log = LogManager.GetCurrentClassLogger();
+
+        /// <summary>
+        /// Gets the property value of the instance.
+        /// </summary>
+        /// <typeparam name="TValue">The type of the value to retrieve.</typeparam>
+        /// <param name="instance">The instance.</param>
+        /// <param name="memberName">The member name.</param>
+        /// <param name="value">The member value to update.</param>
+        /// <returns><c>true</c> if the member was retrieved; otherwise <c>false</c>.</returns>
+        public virtual bool GetMemberValue<TValue>(object instance, string memberName, ref TValue value)
```

Comment:
Why the use of `ref` for TValue instead of `out`? I'm not sure we should encourage the use of two-way behavior with this method as it could introduce confusion to the implementer if they don't understand the nuanced difference between the two.

Most methods implementing a TryGetValue pattern like this return a boolean state and (when supported) a nullable out variable.

It should be up to the implementing caller to decide whether or not the final object is updated based on the return value of `GetMemberValue`. It should be `out TValue? value` and you simply do this for any fallback behavior where not true and avoid the pre-initialization behavior of `ref`:

```
value = null;
return default;